base/abstractdict.jl

```diff
@@ -215,15 +215,40 @@ Dict{Int64,Int64} with 3 entries:
   1 => 0
 ```
 """
-function merge!(combine::Function, d::AbstractDict, others::AbstractDict...)
+function merge!(combine::F, d::AbstractDict, others::AbstractDict...) where F<:Function
```

Comment:
What's the rationale for this change? This is equivalent to what was already written.

Reply:
No, it isn't, this forces specialization on `F` avoiding the non-specialization heuristics.

Reply:
This one cost me quite some pain, by the way, because `@code_native` et al did not warn me that it is showing me the fully realized code, instead of the really existing code (sorry for the pun, cf really existing socialism).